src/value/owned.rs

```diff
@@ -569,13 +601,55 @@ mod value_tests {
             &|e: &OwnedElement| {
                 assert_eq!(Some(&OwnedStruct::from_iter(vec![("greetings", OwnedElement::from(OwnedValue::String("hello".into())))].into_iter())), e.as_struct());
             }
-        )
+        ),
+        // TODO consider factoring this out of the value tests to make it more contained
+        // TODO consider adding non-equivs for this (really symbol token tests are probably better for this)
+        eq_annotations => [
+            // trivially empty is equivalent to another empty
+            vec![vec![], vec![]],
+            // tokens with text
+            vec![
+                vec![text_tok("hello"), text_tok("world")],
+                // containing local sids only
+                vec![
+                    text_tok("hello").with_local_sid(20),
+                    local_sid_tok(21).with_text("world"),
+                ],
+                // mix of local sid, but all with sources
+                vec![
+                    text_tok("hello").with_source("hello_table", 2),
+                    text_tok("world").with_local_sid(59).with_source("world_table", 200)
+                ]
+            ],
+            // tokens without text
+            vec![
+                vec![
+                    // local sid only with no text are all $0 equivalent
+                    local_sid_tok(21),
+                    // source import table is the comparator for unknown cases
+                    local_sid_tok(22).with_source("hello_table", 2)
+                ],
+                vec![
+                    local_sid_tok(0),
+                    local_sid_tok(400).with_source("hello_table", 2),
+                ],
+                vec![
+                    local_sid_tok(200),
+                    text_tok("hello").with_source("hello_table", 2),
+                ],
+                vec![
+                    local_sid_tok(99),
+                    text_tok("hello").with_local_sid(500).with_source("hello_table", 2),
+                ],
+            ],
+        ]
```

Comment:
Really, the symbol token tests should cover this, but it isn't bad to have it here for now.